### PR TITLE
solrbuilder subjects indexer

### PR DESCRIPTION
**Pull request by cdrini**
Work towards: #3317 . Extracted from #4337 .

Feature: Replace the solr-backup shim with a proper solrbuilder subjects indexer!

### Technical
Note: This only works with solr 8, but won't cause solr3 to error if disabled.

### CODE REVIEW

File: scripts/solr_builder/solr_builder/fn_to_cli.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import asyncio
+import typing
+# TODO: BooleanOptionalAction is 3.9 only
+from argparse import ArgumentParser, BooleanOptionalAction, ArgumentDefaultsHelpFormatter, Namespace  # type: ignore
+
+
+class FnToCLI:
+    """Very basic class that converts a simple function into a CLI."""
```
Comment (mekarpeles) [inline scripts/solr_builder/solr_builder/fn_to_cli.py:8]:
```suggestion
    """
    A utility class which automatically infers and generates ArgParse
     command line options from functions based on type annotations
    """
```

File: scripts/solr_builder/solr_builder/index_subjects.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                'json.nl': 'arrarr',
+            })).json()
+    facets = resp['facet_counts']['facet_fields'][f'{subject_type}_facet']
+    docs = [
+        build_subject_doc(subject_type, subject_name, work_count)
+        for [subject_name, work_count] in facets
+    ]
+
+    # TODO: Move commit_way_later into solr_update
+    commit_way_later_dur = 1000 * 60 * 60 * 24 * 5  # 5 days?
```
Comment (mekarpeles) [inline scripts/solr_builder/solr_builder/index_subjects.py:47]: It's would be nice if the value was a var explicitly annotated as `ms`

**Review by mekarpeles — APPROVED** [review]
lgtm, code is a big tough to follow, and some secret params which we could probably make more intuitive, but I trust this works better than our previous solr subjects process (which didn't work) :P

File: scripts/solr_builder/solr_builder/index_subjects.py (hunk 2 above)
Comment (cdrini) [reply scripts/solr_builder/solr_builder/index_subjects.py]: Fixed

**Pull request merged**